packages/gitbeaker-core/src/templates/ResourceDeployTokens.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  BaseRequestOptions,
+  BaseService,
+  BaseServiceOptions,
+  PaginatedRequestOptions,
+  RequestHelper,
+  Sudo,
+} from '../infrastructure';
+
+interface IncludeInherited {
+  includeInherited?: boolean;
+}
+
+type DeployTokenScope = 'read_repository' | 'read_registry' | 'write_registry';
+
+// https://docs.gitlab.com/ee/api/deploy_tokens.html
+export class ResourceDeployTokens extends BaseService {
+  constructor(resourceType: string, options: BaseServiceOptions) {
+    super({ url: resourceType, ...options });
+  }
+
+  add(
+    resourceId: string | number,
+    tokenName: string,
+    tokenScopes: DeployTokenScope[],
+    options?: BaseRequestOptions,
+  ) {
+
+    return RequestHelper.post(this, `${encodeURIComponent(resourceId)}/deploy_tokens`, {
+      name: tokenName,
+      scopes: tokenScopes,
+      ...options,
+    });
+  }
+
+  all({ resourceId, ...options }: { resourceId?: string | number } & PaginatedRequestOptions) {
+
+    if (resourceId) {
+      url = `${encodeURIComponent(resourceId)}/deploy_tokens`;
+    } else {
+      // https://docs.gitlab.com/ee/api/deploy_tokens.html#list-all-deploy-tokens (FIXME?) This needs to be at the root of the webserver, not after the resource type
+      url = 'deploy_tokens';
+    }
+
```

Comment:
This is consistent with other API endpoints. There is no need to do the join below i don't think, and you probably should define this variable